Mathlib/Analysis/InnerProductSpace/Projection/Reflection.lean

```diff
@@ -0,0 +1,168 @@
+/-
+Copyright (c) 2019 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou, Frédéric Dupuis, Heather Macbeth
+-/
+import Mathlib.Analysis.InnerProductSpace.Projection.Basic
+
+/-!
+# Reflection
+
+A linear isometry equivalence `K.reflection : E ≃ₗᵢ[𝕜] E` is constructed, by choosing, for
+each `u : E`, the point `K.reflection u` to satisfy
+`u + (K.reflection u) = 2 • K.orthogonalProjection u`.
```

Comment:
I realize this is pull directly from the existing module documentation, but we should update it to use `starProjection`.
```suggestion
A linear isometry equivalence `K.reflection : E ≃ₗᵢ[𝕜] E` in constructed, by choosing
for each `u : E`, `K.reflection u = 2 • K.starProjection u - u`.
```